### ISSUE TITLE
openssl pkeyutl error when decrypting RSA-OAEP

### ISSUE THREAD

**Issue by licuser** [tpm2-software/tpm2-tools#3292]
Hi,
I am facing a problem when decrypting an RSA-OAEP encrypted data with SHA1. I got error **pkeyutl: Can't set parameter "rsa_padding_mode:oaep":** as described below:
_wrap primary key creation_
`tpm2_createprimary -C o \
                -g sha256 \
                -G rsa \
                -p $PASS \
                -c enroll_rsa.ctx 
HANDLE=$(tpm2_evictcontrol -c enroll_rsa.ctx | cut -d ' ' -f 2 | head -n 1)`
_keypair creation_
`openssl genpkey -provider tpm2 -propquery '?provider=tpm2' \
    -algorithm RSA \
    -pkeyopt bits:2048 \
    -pkeyopt parent:${HANDLE} \
    -pkeyopt parent-auth:$PASS \
    -pkeyopt user-auth:$USER_PASS \
    -out machine.sk.pem`
_Encrypt  data_
`openssl pkeyutl -encrypt -inkey machinepubkey.pem -pubin -in msg.txt -out msg.enc -pkeyopt rsa_padding_mode:oaep -pkeyopt rsa_oaep_md:sha256`
_decrypt data_
`TPM2OPENSSL_PARENT_AUTH=PASSPHRASE openssl pkeyutl -provider tpm2 -provider base -propquery '?provider=tpm2' -inkey /mnt/licpart/enrollement/machine.sk.pem -passin pass:urTx30w5uy04kaD0UEQm -decrypt -in msg.enc -out msg2.txt -pkeyopt rsa_padding_mode:oaep -pkeyopt rsa_oaep_md:sha1

PROVIDER INIT
DER DECODER DECODE
DER DECODER DECODE
TSS2 DECODER DECODE 0x87
TSS2 DECODER LOAD parent: persistent 0x81000000
TSS2 DECODER DECODE 0x87
TSS2 DECODER LOAD parent: persistent 0x81000000
TSS2 DECODER DECODE found RSA
RSA LOAD
RSA GET_PARAMS [ bits security-bits max-size ]
RSA HAS 1
DECRYPT INIT
DECRYPT SET_CTX_PARAMS [ pad-mode ]
pkeyutl: Can't set parameter "rsa_padding_mode:oaep":
RSA FREE
PROVIDER TEARDOWN`

Any help please?

**Comment by JuergenReppSIT** [comment]
@gotthardp Any idea?

**Comment by gotthardp** [comment]
Yeah, the OAEP support is not implemented, see https://github.com/tpm2-software/tpm2-openssl/issues/89.

**Comment by gotthardp** [comment]
I just added support for the OAEP padding, so if you build the latest tpm2-openssl (master branch), your script may work.

**Comment by licuser** [comment]
Perfect, it works fine now!
Thank you
